llvm/lib/Transforms/Utils/CodeLayout.cpp

```diff
@@ -741,12 +750,23 @@ class ExtTSPImpl {
         // Get candidates for merging with the current chain.
         for (const auto &[ChainSucc, Edge] : ChainPred->Edges) {
           // Ignore loop edges.
-          if (ChainPred == ChainSucc)
+          if (Edge->isSelfEdge())
```

Comment:
Nit: This is often called self-loop or loop. If you think fine, I'll push a NFC commit renaming isSelfEdge to isSelfLoop...